### PR TITLE
removing lazy instantiation from autoindexing store

### PR DESCRIPTION
**Pull request by Numbers88s**
Ref #35000 

Removes lazy instantiation from autoindexing store in codeintel

## Test plan
Tested manually, right now it is just stubs. Not actually running any code.

### CODE REVIEW

File: internal/codeintel/autoindexing/internal/store/store.go [hunk 1]
```diff
@@ -68,3 +53,11 @@ const listQuery = `
 SELECT id FROM TODO
 LIMIT %s
 `
+
+// New returns a new autoindexing store.
+func New(db dbutil.DB, observationContext *observation.Context) Store {
```
Comment (efritz) [inline internal/codeintel/autoindexing/internal/store/store.go:58]: Can we add this directly below the struct def?

File: internal/codeintel/autoindexing/internal/store/store.go [hunk 2]
```diff
@@ -14,42 +14,27 @@ import (
 	"github.com/sourcegraph/sourcegraph/lib/errors"
 )
 
-type Store struct {
-	*basestore.Store
-	operations *operations
-}
+// ensure that all the needed methods are implemented
+var _ Store = (*store)(nil)
```
Comment (efritz) [inline internal/codeintel/autoindexing/internal/store/store.go:18]:
```suggestion
```

The New function already enforces this.

File: internal/codeintel/autoindexing/init.go [hunk 3]
```diff
@@ -24,29 +24,25 @@ var (
 
 func GetService(db database.DB) *Service {
 	svcOnce.Do(func() {
-		observationContext := &observation.Context{
+		storeObservationCtx := &observation.Context{
+			Logger:     log.Scoped("store.autoindexing", "autoindexing store"),
```
Comment (efritz) [inline internal/codeintel/autoindexing/init.go:28]: Can we flip this to `autoindexing.store`? Just realized they don't match.

**Review by efritz — APPROVED** [review]

File: internal/codeintel/autoindexing/internal/store/store.go (hunk 2 above)
Comment (Numbers88s) [reply internal/codeintel/autoindexing/internal/store/store.go:18]: Removed!

File: internal/codeintel/autoindexing/internal/store/store.go (hunk 1 above)
Comment (Numbers88s) [reply internal/codeintel/autoindexing/internal/store/store.go:58]: Yeah no problem. I tend to put it at the end cause we only use it once but we see it everytime we open the file lol. It's all good, I put right after the struct :v:

File: internal/codeintel/autoindexing/init.go (hunk 3 above)
Comment (Numbers88s) [reply internal/codeintel/autoindexing/init.go:28]: Fixed.

**Pull request merged**